src/sql/workbench/contrib/executionPlan/browser/executionPlanComparisonEditorView.ts

```diff
@@ -640,6 +666,35 @@ class PropertiesAction extends Action {
 	}
 }
 
+enum PlanIdentifier {
+	Primary = 0,
+	Added = 1
+}
+
+class SearchNodeAction extends Action {
+	public static ID = 'epCompare.searchNodeAction';
+	public static LABEL = localize('epCompare.searchNodeAction', 'Find Node');
+	public static LABEL_FOR_ADDED_PLAN = localize('epCompare.searchNodeActionAddedPlan', 'Find Node - Added Plan');
+
+	constructor(private readonly _planIdentifier: PlanIdentifier, @IAdsTelemetryService private readonly telemetryService: IAdsTelemetryService) {
+		const getLabelForAction = () => {
+			return _planIdentifier === PlanIdentifier.Added ? SearchNodeAction.LABEL_FOR_ADDED_PLAN : SearchNodeAction.LABEL;
+		};
+
+		super(SearchNodeAction.ID, getLabelForAction(), searchIconClassNames);
+		this.enabled = false;
+	}
+
+	public override async run(context: ExecutionPlanComparisonEditorView): Promise<void> {
+		let executionPlan = this._planIdentifier === PlanIdentifier.Added ? context.activeBottomPlanDiagram : context.activeTopPlanDiagram;
+		let widgetController = this._planIdentifier === PlanIdentifier.Added ? context.bottomWidgetController : context.topWidgetController;
+
+		this.telemetryService.sendActionEvent(TelemetryKeys.TelemetryView.ExecutionPlan, TelemetryKeys.TelemetryAction.FindNode);
+		let nodeSearchWidget = context._instantiationService.createInstance(NodeSearchWidget, widgetController, executionPlan);
```

Comment:
You shouldn't be using services from other objects like this - instead inject it into the constructor of this action